ecs-cli/modules/cli/local/secrets/secrets.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2015-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Package secrets implements functions to decrypt container secrets defined in a ECS task definition.
+package secrets
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/pkg/errors"
+)
+
+// SecretDecrypter wraps the DecryptSecret function.
+//
+// DecryptSecret returns the decrypted value of a secret given its ARN.
+type SecretDecrypter interface {
+	DecryptSecret(arn string) (string, error)
+}
+
+// A ContainerSecret labels an ECS secret with the container name it belongs to.
+type ContainerSecret struct {
+	containerName string
+	secret        ecs.Secret
+}
+
+// NewContainerSecret returns a new container secret.
+func NewContainerSecret(containerName, secretName, secretValue string) *ContainerSecret {
+	return &ContainerSecret{
+		containerName: containerName,
+		secret: ecs.Secret{
+			Name:      aws.String(secretName),
+			ValueFrom: aws.String(secretValue),
+		},
+	}
+}
+
+// Decrypt returns the decrypted secret value.
+func (cs *ContainerSecret) Decrypt(sd SecretDecrypter) (string, error) {
+	return sd.DecryptSecret(aws.StringValue(cs.secret.ValueFrom))
+}
+
+// ServiceName returns whether the secret belongs to SSM or Secrets Manager.
+// If it can't determine the service or the ARN belongs to a different service then it returns an error.
+func (cs *ContainerSecret) ServiceName() (string, error) {
+	secretARN := aws.StringValue(cs.secret.ValueFrom)
+	parsedARN, err := arn.Parse(secretARN)
+	if err != nil {
+		if strings.Contains(err.Error(), "arn: invalid prefix") {
```

Comment:
Good question! I tried to explain it in the comment inside this if-block.The ECS Task Definition also accepts a value that's not an ARN which is just the parameter name in SSM (see https://docs.aws.amazon.com/AmazonECS/latest/developerguide/specifying-sensitive-data.html#secrets-logconfig). So this function will also default to SSM.

If it's just an invalid ARN then we will try to look it up in SSM and the request will fail, so the user will receive an error while trying to start the container with `local up`. Does that help?

Reply:
I see. So SSM is the default one you set. Then, that will make sense! Since there are two types of return error defined in `arn.Parse`, it will also possibly error out because there are more than 6 sections in the ARN rather than failing the request when you return [this](https://github.com/aws/amazon-ecs-cli/blob/2e80b3db2f189c59f377907249eef7516e4c445b/ecs-cli/modules/cli/local/secrets/secrets.go#L70).